### PR TITLE
Fix python 3 compatibility.

### PR DESCRIPTION
**Pull request by xernobyl**
https://trello.com/c/h7UG7vS3/1033-make-api-work-on-both-python-2-and-3

### CODE REVIEW

File: valispace/__init__.py [hunk 1]
```diff
@@ -59,25 +65,39 @@ def __init__(self, url=None, username=None, password=None):
...
+			raise Exception("VALISPACE-ERROR: " + str(e))
 
 		except:
 			# TODO:
 			# T: Capture specific exceptions, bc it is also possible that
 			# the endpoint does not work or something like that...
 			raise Exception("VALISPACE-ERROR: Invalid credentials or url.")
 
+		json = result.json()
+		print('json:', json)	# debug
```
Comment (lollolindblad) [inline valispace/__init__.py]: Remove debug prints.

File: valispace/__init__.py [hunk 2]
```diff
@@ -162,7 +182,14 @@ def get_vali_list(self, workspace_id=None, workspace_name=None, project_id=None,
 		if vali_marked_as_impacted:
 			url = self.__increment_url(url) + "valis_marked_as_impacted={}".format(vali_marked_as_impacted)
 		response = requests.get(url, headers=self.get_request_headers())
-		return response.json()
+
+		if response.status_code != 200:
```
Comment (lollolindblad) [inline valispace/__init__.py:185]: Are these prints for debug or should it stay in the code?

File: valispace/__init__.py [hunk 3]
```diff
@@ -306,7 +333,15 @@ def get_component_list(self, workspace_id=None, workspace_name=None, project_id=
 		elif tag_name:
 			url = self.__increment_url(url) + "tags__name={}".format(tag_name)
 
-		return requests.get(url, headers=self.get_request_headers()).json()
+		response = requests.get(url, headers=self.get_request_headers())
+
+		if response.status_code != 200:
```
Comment (lollolindblad) [inline valispace/__init__.py:337]: Are these prints for debug or should it stay in the code?

**Review by lollolindblad — CHANGES_REQUESTED** [review]

File: valispace/__init__.py (hunk 1 above)
Comment (xernobyl) [reply valispace/__init__.py]: This one wasn't supposed to be in here.

File: valispace/__init__.py (hunk 2 above)
Comment (xernobyl) [reply valispace/__init__.py:185]: These prints can be left here, I think.

File: valispace/__init__.py (hunk 3 above)
Comment (xernobyl) [reply valispace/__init__.py:337]: These prints can be left here, I think.

**Review by lollolindblad — APPROVED** [review]

**Pull request merged**